test/integration/consul-container/test/upgrade/peers_http_test.go

```diff
@@ -49,7 +49,9 @@ func TestPeering_UpgradeToTarget_fromLatest(t *testing.T) {
 		acceptingClient, err := acceptingCluster.GetClient(nil, false)
 		require.NoError(t, err)
 
-		_, gatewayAdminPort := dialing.Gateway.GetAdminAddr()
+		gatewayProxy, ok := dialing.Gateway.(*libservice.GatewayContainer)
```

Comment:
I think dialing.Gateway is a `libservice.Service` interface so there's no need to cast the type. Is there anything changed in this PR.

Reply:
explained above, `GetAdminAddr` should only be called from GatewayContainer and ConnectContainer types.